### DISCUSSION TITLE
magit-blame shows CommitDate instead of AuthorDate

### DISCUSSION THREAD

**Discussion by dadinn** [magit/magit#5080] [Ideas]
### Problematic behaviour
`magit-log` correctly displays `AuthorDate` for commits, yet for some reason using `magit-blame-addition` on a buffer displays the `CommitDate` instead. This is confusing and counter-intuitive.

The `CommitDate` should be only relevant for auditing who and when amended/rebased a commit.

### Expected behaviour

The `AuthorDate` to be shown on the blame interface.

### Reproduce

Run `magit-blame-addition` on a version controlled buffer with a history of rebases, amendments, and/or rewordings on past commits.

### Magit version

Checked with most recent commit:
`f4ff817cb2a48f0f7887050c3be469c03a059567`

**Comment by tarsius** [comment]
> The `CommitDate` should be only relevant for auditing who and when amended/rebased a commit.

I disagree. It is important to be able to tell which commit comes first, based solely on the date. The author date does not allow that because the commits may have been reordered.

But we don't have to agree on this, you can customize `magit-blame-styles` to use the author date instead.

**Comment by dadinn** [comment]
It is indeed true that the commits might have been reordered, but that's the most extreme case which completely breaks any timestamp based assumptions, and even the `CommitDate` won't make much sense. Except maybe that one could see which commits in the past were not affected by such changes. So I see your point.

Nevertheless, with using the `AuthorDate`, at least one would see the same date on the blame buffer for the same commit, as on the magit log (which uses the `AuthorDate`). That's why I said I consider `CommitDate` is for audit. If the log shows older dates are preceded by newer ones, then one could look at the `CommitDate` and see that there was a more recent rebase, and someone must have reordered them.

I've tried changing `heading-format` in the `magit-blame-styles`. It worked perfectly, thanks!

**Reply by dadinn** [reply]
Just thought I'll put here the code which worked for me:
```
(let* ((headings (alist-get (quote headings) magit-blame-styles))
       (heading-format (assoc (quote heading-format) headings)))
  (setcdr heading-format "%-20a %A %s\n"))
```
